### PR TITLE
User similarity

### PR DESCRIPTION
**Pull request by mayhem**
This PR is the user similarity work that lucifer and I have been working on, based on Param's work.

### CODE REVIEW

File: admin/create-dumps.sh
Comment (mayhem) [inline admin/create-dumps.sh:35]: Sorry for the unrelated fix here -- I think this will be replaced by another PR possibly before this is even merged.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recording/user_similarity.py:25]: types and docstrings on these methods would really help readability.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recording/user_similarity.py:29]: style thing, but maybe we should consider extracting this logic for getting max and min similarity into a separate function. Or at least a comment here explaining that's what we're doing.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (paramsingh) [inline listenbrainz_spark/recommendations/recording/user_similarity.py:76]: style thing again, but comments in this and the next few lines would greatly enhance readability in my opinion. It's not obvious at the outset what a co-ordinate matrix or a indexed-row-matrix or vectors_mapped_rdd is

File: listenbrainz_spark/query_map.py
Comment (mayhem) [reply listenbrainz_spark/query_map.py:39]: I think because we're still using recording data to calculate this.... But I'll let lucifer chime in...

File: listenbrainz_spark/recommendations/recording/create_dataframes.py
Comment (mayhem) [reply listenbrainz_spark/recommendations/recording/create_dataframes.py:350]: Ideally yes, but I feel that there is quite a bit of cleanup to do in the spark code along these lines and I don't feel now is the time to pick it all up.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (mayhem) [reply listenbrainz_spark/recommendations/recording/user_similarity.py:29]: I don't really feel that this is needed -- this code is pretty straightforward.

File: listenbrainz_spark/query_map.py
Comment (amCap1712) [reply listenbrainz_spark/query_map.py:39]: Yes, I intend to move user_similarity to a different package. I initially put it there to allow easy reuse of code. Now that we have a working implementation, I'll do the cleanup (add comments, type annotations, refactor code) and add tests.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (mayhem) [reply listenbrainz_spark/recommendations/recording/user_similarity.py:35]: I added some comments which should help here.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/user_similarity.py]: Done.

File: listenbrainz_spark/recommendations/recording/user_similarity.py
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/user_similarity.py]: I have added some description in the docstring.

File: listenbrainz_spark/recommendations/recording/create_dataframes.py
Comment (amCap1712) [reply listenbrainz_spark/recommendations/recording/create_dataframes.py]: Done.

**Review by paramsingh — APPROVED** [review]
looks good to me, thanks for making those changes!

**Pull request merged**
